2_process.R

```diff
@@ -40,13 +51,15 @@ p2_targets_list <- list(
        get_site_flowlines(p1_reaches_sf, sites_tbl, sites_crs = 4269, max_matches = 1, search_radius = 0.1)
      }),
   
-  # Aggregate instantaneous SC data to daily min/mean/max
-  tar_target(
-    p2_inst_data_daily,
-    aggregate_data_to_daily(p1_inst_data,p1_daily_data, min_daily_coverage=0.5, output_tz="America/New_York")),
-  
-  # Combine 1) daily DO data and 2) instantaneous DO data that has been aggregated to daily 
+  # Pair PRMS segments with NHDPlusV2 reaches
   tar_target(
-    p2_daily_combined,
-    bind_rows(p1_daily_data, p2_inst_data_daily))
+    p2_prms_nhdv2_xwalk,
+    {
+      p1_reaches_sf %>%
+        split(.,.$subsegid) %>%
+        purrr::map(.,pair_nhd_reaches,nhd_lines=p1_nhdv2reaches_sf) %>%
```

Comment:
same question as above re `purrr` instead of `targets` dynamic branching.

Reply:
I used `purrr` here because as far as I can tell, I cannot apply `targets::map` and dynamic branching to an sf spatial object. I've therefore opted to leave this code for `p2_prms_nhdv2_xwalk` as is, but I can revisit this target if I find out later that branching over spatial objects is possible.